addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/handler/YamahaBridgeHandler.java

```diff
@@ -0,0 +1,283 @@
+    private Logger logger = LoggerFactory.getLogger(YamahaBridgeHandler.class);
+    private int refrehInterval = 60; // Default: Every 1min
+    private float relativeVolumeChangeFactor = 0.5f; // Default: 0.5 percent
+    private ScheduledFuture<?> refreshTimer;
+    private ZoneDiscoveryService zoneDiscoveryService;
+
+    private HttpXMLSendReceive xml;
+    private SystemControl.State state = new SystemControl.State();
+
+    public YamahaBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    public float getRelativeVolumeChangeFactor() {
+        return relativeVolumeChangeFactor;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (xml == null || state.host == null) {
+            return;
+        }
+
+        String id = channelUID.getId();
+
+        try {
+            if (command instanceof RefreshType) {
+                refreshFromState(channelUID);
+                return;
+            }
+
+            switch (id) {
+                case YamahaReceiverBindingConstants.CHANNEL_POWER: {
+                    boolean oldState = state.power;
+                    SystemControl basicDeviceInformation = new SystemControl();
+                    basicDeviceInformation.setPower(xml, ((OnOffType) command) == OnOffType.ON, state);
+                    if (!oldState && state.power) {
+                        // If the device was off and now turns on, we trigger a refresh of all zone things.
+                        // The user might have renamed some of the inputs etc.
+                        updateAllZoneInformation();
+                    }
+                }
+                    break;
+                default:
+                    logger.error("Channel " + id
+                            + " not supported on the yamaha device directly! Try with the zone things instead.");
+            }
+        } catch (Exception e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    private void refreshFromState(ChannelUID channelUID) {
+        switch (channelUID.getId()) {
+            case YamahaReceiverBindingConstants.CHANNEL_POWER:
+                updateState(channelUID, state.power ? OnOffType.ON : OnOffType.OFF);
+                break;
+            default:
+                logger.error("Channel refresh for " + channelUID.getId() + " not implemented!");
+        }
+    }
+
+    /**
+     * Sets up a refresh timer (using the scheduler) with the CONFIG_REFRESH interval.
+     *
+     * @param initial_wait_time The delay before the first refresh. Maybe 0 to immediately
+     *            initiate a refresh.
+     */
+    private void setupRefreshTimer(int initial_wait_time) {
+        if (state == null) {
+            return;
+        }
+
+        Object interval_config_o = thing.getConfiguration().get(YamahaReceiverBindingConstants.CONFIG_REFRESH);
```

Comment:
To have a constant naming scheme please use camel casing instead of underlines. Please change the names of other variables, too.